lib/elixir/lib/enum.ex

```diff
@@ -2259,14 +2266,12 @@ defmodule Enum do
     do: []
   def take_random(first..first, count) when is_integer(count) and count >= 1,
     do: [first]
-  def take_random(first..last, 1) when first > last,
-    do: take_random(last..first, 1)
-  def take_random(first..last, 1),
-    do: [random_index(last - first) + first]
+  def take_random(_.._ = range, 1),
+    do: [random(range)]
```

Comment:
The new implementation should just call random_integer(...) the amount of times desired. :) Something like:

```elixir
def take_random(_.._ = range, n) do
  take_random_range(range, n)
end

defp take_random_range(_range, 0), do: 0
defp take_random_range(range, n), do: [random(range) | take_random_range(range, n - 1)]
```